runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessConfigureView/ProcessCredentials/ProcessCredentials.tsx

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { Typography } from '@mui/material';
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+import { useSnackbar } from 'notistack';
+
+import LockIcon from '#public/images/icons/lock.svg';
+import { useRequiredCredentialTypes } from '#src-app/credentials/useRequiredCredentialTypes';
+import useTranslations from '#src-app/hooks/useTranslations';
+import useWindowSize from '#src-app/hooks/useWindowSize';
+import { useDispatch, useSelector } from '#src-app/store';
+import { processActions, processSelector } from '#src-app/store/slices/Process';
+
+import ActionCredential from './ActionCredential';
+import ActionCredentialAdd from './ActionCredentialAdd';
+import {
+    ActionBox, ActionBoxContent,
+    ActionBoxHeader, ActionsColumns,
+    ActionsContainer, Container, Header
+} from './ProcessCredentials.styles';
+import { ActionSortedColumns, sortByActionCredentialType, sortByColumns } from './ProcessCredentials.utils';
+import { ProcessCredentialsAddDialog } from './ProcessCredentialsAddDialog';
+import { ProcessCredentialsDeleteDialog } from './ProcessCredentialsDeleteDialog';
+
+const ACTION_MIN_WIDTH = 400;
+const MARGIN_LIMIT = 800;
+
+const ProcessCredentials = () => {
+    const { enqueueSnackbar } = useSnackbar();
+    const { translate }= useTranslations();
+    const dispatch = useDispatch();
+    const { id: processId } = useRouter().query;
+    const { draft: { credentials: processCredentials } } = useSelector(processSelector);
+    const credentialTypes = useRequiredCredentialTypes();
+    const actionCredentials = useMemo(
+        () => credentialTypes ? sortByActionCredentialType(processCredentials, credentialTypes) : null,
+        [processCredentials, credentialTypes]
+    );
+
+    const [columns, setColumns] = useState<ActionSortedColumns>([]);
+    const [isAddDialogOpen, setIsAddDialogOpen] = useState(false);
+    const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+    const [currentCredentialId, setCurrentCredentialId] = useState<string | null>(null);
```

Comment:
I think there should be created type for CredentialId = string | null. Missing it is an antipattern of primitive obsession. There's a risk that if we change it to string | null | undefined (which I think can happen) there'll be a problem with changing it across the whole project

Reply:
I will do it